src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/old_gen/LevelTwoActionBuilder.java

```diff
@@ -0,0 +1,183 @@
+        .readQueueEWMASize(nodeConfigCache, esNode, ResourceEnum.SEARCH_THREADPOOL);
+    if (writeQueueEWMASize == null || searchQueueEWMASize == null) {
+      return;
+    }
+    ModifyQueueCapacityAction writeQueueAction = queueActionMap.get(ResourceEnum.WRITE_THREADPOOL);
+    ModifyQueueCapacityAction searchQueueAction = queueActionMap.get(ResourceEnum.SEARCH_THREADPOOL);
+    if (writeQueueAction != null && searchQueueAction != null) {
+      int writeQueueSizeBucket = bucketization(writeQueueAction.getLowerBound(),
+          writeQueueAction.getUpperBound(), writeQueueEWMASize, LEVEL_TWO_CONST.QUEUE_SIZE_BUCKET_SIZE);
+      int searchQueueSizeBucket = bucketization(searchQueueAction.getLowerBound(),
+          searchQueueAction.getUpperBound(), searchQueueEWMASize, LEVEL_TWO_CONST.QUEUE_SIZE_BUCKET_SIZE);
+      if (writeQueueSizeBucket > searchQueueSizeBucket) {
+        actionFilter.put(ResourceEnum.WRITE_THREADPOOL, true);
+      }
+      else if (writeQueueSizeBucket < searchQueueSizeBucket) {
+        actionFilter.put(ResourceEnum.SEARCH_THREADPOOL, true);
+      }
+      //tie breaker. default policy prefer writing over indexing
+      else {
+        actionFilter.put(ResourceEnum.WRITE_THREADPOOL, true);
+      }
+    }
+    else if (writeQueueAction != null) {
+      actionFilter.put(ResourceEnum.WRITE_THREADPOOL, true);
+    }
+    else if (searchQueueAction != null) {
+      actionFilter.put(ResourceEnum.SEARCH_THREADPOOL, true);
+    }
+  }
+
+  @Override
+  protected void registerActions() {
+    addFieldDataCacheAction();
+    addShardRequestCacheAction();
+    addSearchQueueAction();
+    addWriteQueueAction();
+  }
+
+  /**
+   * The default priority in this level is
+   * 1. downsize both caches simultaneously with larger step size.
+   * 2. Allocate size of each queue into buckets and downsize any queue that has
+   * higher bucket size.
+   * 3. if the bucket size of both queues are equal,
+   * the action priority will act as tie breaker and the default setting
+   * favors write over search as write is more important in log analytic workload
+   * 4. The above default action priority can be overridden by customer settings
+   * but customer's acton priority will only affect step 3 above. Step 1 and 2 will
+   * be executed regardless of priority action settings
+   */
+  // TODO : read priority from yml if customer wants to override default ordering
+  @Override
+  protected void actionPriorityFilter() {
+    actionPriorityForCache();
+    actionPriorityForQueue();
+  }
+
+  //TODO : read consts from rca.conf
+  private static class LEVEL_TWO_CONST {
+    public static final double FIELD_DATA_CACHE_LOWER_BOUND = 0.02;
+    public static final double SHARD_REQUEST_CACHE_LOWER_BOUND = 0.01;
+    public static final int CACHE_ACTION_STEP_COUNT = 2;
+    public static final int QUEUE_ACTION_STEP_COUNT = 1;
+    public static final int QUEUE_SIZE_BUCKET_SIZE = 10;
```

Comment:
Should this be a separate constant? These are just bands that you are using to divide the range of queue capacity, can this just be local to the `bucketization` function?